section3/reflection.md

```diff
@@ -1,13 +1,14 @@
 ## Section 3 Reflection
 
-1. In Section 3, we removed even more supports and pushed you to learn a bit more independently, read through technical documentation, and use your resources more. How did that feel?
-
-2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.
-
-3. In your own words, explain what an Array method is.
-
-4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
-
-5. If you had to teach a 5 year old about loops, how would you explain it?
-
-6. What questions do you still have about Arrays and loops?
+1. **In Section 3, we removed even more supports and pushed you to learn a bit more independently, read through technical documentation, and use your resources more. How did that feel?**
+I think it has been an appropriate, gradual removal of supports. It encourages me to consistently refresh my memory with concepts that I may have forgotten from a previous section. For example, having us integrate if...else statements and concatenation into loops strengthens my memory of those concepts.
+2. **Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.**
+Developers probably use arrays in Google Maps to store recent locations you searched for. This data is pulled up whenever we click the search bar or when we are picking a location to arrive at or depart from. This makes it quick and simple to use previous locations searched for.
+3. **In your own words, explain what an Array method is.**
+An Array method allows us to alter our array without having to write a brand new array from scratch. We can remove, insert, shuffle, and alter items around with a quick function.
+4. **What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?**
+A loop is used to repeat something, such as a console.log message, repeatedly without having to type it out every time. We use a loop in conjunction with an array when wanting to run a repeated code for each item in an array. Loops don't necessarily have to be used in arrays that are smaller or if we don't plan on running a repeated code.
+5. **If you had to teach a 5 year old about loops, how would you explain it?**
+When you drink water from a water fountain at school, each time you drink it, you have to take your hands and press on the big button to get water come up. A loop is similar to making that water fountain know that you want water because you're standing right by it. You make the water fountain automatically give you water each time.
+6. **What questions do you still have about Arrays and loops?**
+I saw that there are other types of loops and am curious to see how those integrate into what we've already learned. I am still having trouble understanding arrays, especially when put together with the if...else statements.
```

Comment:
Yes, there are many other tools we can use to loop or iterate over arrays! You'll get into them in Mod 1. They all are built on top of the concept of the for loop, so having a good understanding of that is most important now.

RE "when put together with the if/else" - I think that hits on the idea of nested logic or nested code blocks. Ex: an if statement inside of a loop. Those can definitely be tricky - from the logic standpoint as well as syntax. If you are writing nested code, one great strategy is to annotate each line, to force yourself to slow down and ensure you understand it's role.